main.cpp

```diff
@@ -69,46 +69,497 @@ write 3 UDTs below that EACH have:
+        nadine.numberOfElbows = 0;
+    }
+}
+
+float Cabin::reclineSeat(float maxReclineAngle, bool rearPassengerIsHappy)
+{
+    float angle = 0.0f;
 
+    while(angle < maxReclineAngle)
+    {
+        if(angle > 30.0f)
+        {
+            rearPassengerIsHappy = false;
+        }
+
+        if(rearPassengerIsHappy)
+        {
+            std::cout << "Recline seat" << std::endl;
+            angle += 5.0f;
+        }
+        else
+        {
+            std::cout << "No recline" << std::endl;
+            return angle;
+        }
+    }
+    return angle;
+}
 /*
- copied UDT 3:
- */
+copied UDT 3:
+*/
+struct Iphone
+{
+    Iphone();
+
+    std::string operatingSystem;
+    float screenSize{8.0f};
+    std::string speakerType;
+    std::string cameraType;
+    float batteryLife = 100.0f;
+
+    void browseTheWeb(std::string url);
+    void playMusic(std::string songName);
+    void makePhoneCall(int number);
+    void screenShrinker(int magnitude);
+};
 
+Iphone::Iphone():
+operatingSystem("iOS"),
+speakerType("Bose"),
+cameraType("Nikon")
+{   
+    std::cout << "Constructing IPhone" << std::endl;
+}
+
+void Iphone::browseTheWeb(std::string url)
+{   
+    operatingSystem = speakerType + cameraType;
+    std::cout << "www." << url << std::endl;
+}
+
+void Iphone::playMusic(std::string songName)
+{
+    screenSize = batteryLife;
+    std::cout << speakerType << " " << songName << std::endl;
+}
+
+void Iphone::makePhoneCall(int number)
+{
+    std::cout << "Listen on on " << speakerType << " speakers" << std::endl;
+    number += 1;
+}
+
+void Iphone::screenShrinker(int magnitude)
+{
+    int i = magnitude % 10;
+    while(i < 10)
+    {
+        screenSize -= 0.1f;
+        ++i;
+    }
+}
 /*
- new UDT 4:
- with 2 member functions
- */
+new UDT 4:
+with 2 member functions
+*/
+struct WishList
+{
+    WishList();
+    ~WishList();
+
+    Cabin myCabin;
+    Cabin::FlightAttendant jerry;
+    Diner myDiner;
+    Diner::Kitchen myKitchen;
+    Iphone myPhone;
+
+    void playOnPhone();
+    std::string setMood(float lightLevel);
+};
+
+WishList::WishList()
+{
+    myCabin.serveCoffee(jerry);
+    std::cout << "Construct WishList. Coffee is served!" << std::endl;
+}
+
+WishList::~WishList()
+{
+    std::cout << "Deconstruct WishList. Please recycle!" << std::endl;
+}
+
+void WishList::playOnPhone()
+{
+    myPhone.batteryLife -= 10.0f;
+}
+
+std::string WishList::setMood(float lightLevel)
+{
+    myCabin.dimLights(lightLevel);
+    myCabin.playMusic();
+    return "Lights dimmed to " + std::to_string(lightLevel) + " and music playing";
+}
 
 /*
- new UDT 5:
- with 2 member functions
- */
+new UDT 5:
+with 2 member functions
+*/
+struct SearchEngine
+{
+    SearchEngine();
+    ~SearchEngine();
+
+    Cabin myCabin;
+    Cabin::FlightAttendant jerry;
+    Diner myDiner;
+    Diner::Kitchen myKitchen;
+    Iphone myPhone;
+
+    void animateFlightAttendant(float insanityLevel);
+    std::string prepEggs(int tableNumber);
+};
```

Comment:
How are any of these member variables related to this new UDT?
- [ ] revise, make it make sense